content/blogs/Porting-Syzkaller-to-ReactOS-GSoC-2020.md

```diff
@@ -0,0 +1,32 @@
+---
+title:       "Porting Syzkaller to ReactOS - GSoC 2020"
+author:      "Freakston"
+date:        2020-05-27
+tags:        [ gsoc ]
+---
+
+Hey , I am Suraj K Suresh (@Freakston) , one of the GSoC students of ReactOS. I'm very excited to spend my summer contributing to ReactOS.  
+
+This is the first time I've applied for GSoC and my mentors for the project are Victor Perevertkin and Timo Kreuzer.
+
+My project is "Porting Syzkaller to ReactOS". The project is getting one of the most famous Linux fuzzers to fuzz ReactOS. The project will lead to improvement of  the kernel codebase by detecting code crashes.
+
+## My Introduction to ReactOS
+
+I first stumbled upon ReactOS while browsing Hackernews. I found a post containing information about ReactOS which piqued my curiosity on the topic. After I delved deeper into the subject I became further interested and then I joined the community. After spending some time going through the codebase and having further discussions, I decided on applying for GSoC.
+
+## Details about the project
+
+Syzkaller is an **unsupervised coverage-guided kernel fuzzer** by Google. The main goal of the fuzzer is to find and eliminate incorrect or malicious code from the kernel and help in improving the code quality. Syzkaller uses the kernel syscall API to interact and execute kernel code from the userspace.
+
+The first few weeks of the community bonding period were spent going through the implementation and codebase of Syzkaller. 
+
+Next step was getting Syzkaller running on Windows x64bit. The current master of Syzkaller isn't maintained anymore for Windows.  The codebase has been refactored several times without testing on Windows. So moving back a few commits to the Initial support for Windows, we have a working build of Syzkaller for Windows x64bit :)
+
+Now the next step is adding the i386 arch for Windows. Making the necessary changes, we now have the components working on x86 bit of Windows. 
+
+Finally we get to the ReactOS part. ReactOS reports NT 5.2 which is around Windows XP and Server 2003. Syzkaller uses Slim Reader/Writer locks  which was first implemented in the Windows Vista kernel. After making our kernel export the needed functions (mainly sync functions) we have the components running.
```

Comment:
The edition of Windows XP that reports as NT 5.2 build internally is Windows XP Professional x64 Edition, just for clarification. Other than that it looks good to me.

Reply:
Do you want me to change it ? @Fraizeraust

Reply:
Yes.

Reply:
Done

Reply:
Well what I exactly mean by my comment is the Windows XP build that reports as NT 5.2 is Professional x64 Edition which you did not explicitly mention what build of Windows XP reports as such. It was not necessary to remove Windows Server 2003 because that's correct.

If that brought you confusions then apologies.

Reply:
No problem I guess I confused myself by reading it wrong :)